core/src/main/scala/io/finch/request/package.scala

```diff
@@ -162,6 +180,8 @@ package object request {
       def apply[Req](req: Req)(implicit ev: Req => HttpRequest) =
         try number.toFuture
         catch { case _: IllegalArgumentException => ValidationFailed(param, rule).toFutureException }
+
+      override def toString: String = s"param: $param"
```

Comment:
This sounds like a big amount of work. But I'm already in love with this feature. @jenshalm do you mind to take care of this task? I think it will be fair, since it's originally your idea. If so, please, consider splitting the `package.scala` file into `RequestReader.scala` and `ValidationRule` since we have two big basic blocks now. @rpless do you mind if Jens continue working on it?